### DISCUSSION TITLE
[2023-02-20] Performance Team

### DISCUSSION THREAD

**Discussion by wdanilo** [enso-org/enso#5695] [Weekly Demo Day]
1. #5224 was fixed.
2. #5016 was implemented.
3. The Electron app should have some important performance upgrades on some platforms and should work on more systems by default now.
4. I also implemented [cargo-watch-plus](https://github.com/enso-org/cargo-watch-plus), a better version of `cargo-watch` that works on modern MacOS (yay, I can use watch again!), and is more powerful than `cargo-watch` (for example, it supports all `.gitignore` syntax!).
5. I started working on https://github.com/enso-org/enso/issues/5697 with some good progress already.

**Comment by JaroslavTulach** [comment]
Four topics by the engine-ers:
- #5698 - prepared, analyzed and created by Dmitry - adopted by Jaroslav. Work goes on.
- #5700 - PoC by Hubert, discussions, first encouraging results. Work will go on.
- Investigated, analyzed, and mitigated **slowdown of comparators** after their integration.
    - _Good news_: Pavel has a fix: #5687
    - **Bad news**: `if_then_else` _"cascades"_ are slow and it has to be fixed. Work will be planned.
- #5680 - Implemented, tested, integrated. ~10% of startup work now be done in parallel.

Various stability fixes in addtion to that.

**Comment by kazcw** [comment]
- Precompile shaders (#5606): Just reviewed, merging today.
- Profile frontend (#5603): Most time is now accounted for.
  - We are spending 23% of our time rendering text, about half of that in MSDF calls.
  - We are spending 23% of our time parsing documentation.
- Optimize dropdown creation (#5688): Cut opening time in half.
- This week: Investigate/improve MSDF usage (#5669).
  - Is the cache working?
  - Can/should we pre-seed the cache?
- Then: Doc parsing.
